### PR TITLE
Finalize entrypoint refactor

### PR DESCRIPTION
**Pull request by scruffystuffs**
## Cleanup tasks

Many of these tasks are the nits/changes needed for #738, #742, #754 

- [x] Move `NewFossa` Modules into `Fossa`.
- [x] Fix validation error reporting
- [x] Add `fossa analyze --monorepo`
- [x] Add `fossa log4j`
- [x] Rename `GlobalOpts` to `CommonOpts`
- [x] Move `ConfigFile` And `EnvironmentVars` to `App.Fossa.Config` namespace.
- [x] ~Use config type in debug bundles.~ This was left as a TODO in the code, we can fix it later.

## Notes

Validation error reporting example:

image

### CODE REVIEW

**Review by cnr — APPROVED** [review]
We'll need to do way more manual testing before merging this into master, but this particular diff seems pretty safe/small

The issues I found were undiscovered from other PRs, and were a little scary. I'll do a bigger review/sweep of the wider `master..HEAD` diff in the next couple of days, because we should get this merged before it grows much bigger

File: src/App/Fossa/Config/Container.hs [hunk 1]
```diff
@@ -239,8 +232,8 @@ data ContainerScanConfig
 
 instance GetSeverity ContainerCommand where
```
[the remark this answers is not in the document]
Comment (scruffystuffs) [reply src/App/Fossa/Config/Container.hs:233]: I'd love a better solution for this, I just don't know what that would look like.

File: src/Control/Effect/Diagnostics.hs [hunk 2]
```diff
@@ -134,20 +135,28 @@ recover' = send . Recover'
...
 errorBoundary = send . ErrorBoundary
 
-type Validator = Validation (NonEmpty FailureBundle)
+-- | A convenience type for the 'Validation' returned by 'validationBoundary'.
+type Validator = Validation (NonEmpty SomeDiagnostic)
 
+-- | Same as 'errorBoundary', but converted to a 'Validator'.
 validationBoundary :: Has Diagnostics sig m => m a -> m (Validator a)
-validationBoundary act = eitherToValidation . first (NE.:| []) <$> errorBoundary act
+validationBoundary act = eitherToValidation . first (toNEList . failureCause) <$> errorBoundary act
```
[the remark this answers is not in the document]
Comment (scruffystuffs) [reply src/Control/Effect/Diagnostics.hs:151]: Not in the version of `base` that we're using.

[the remark this answers is not in the document]
Comment (scruffystuffs) [reply src/Control/Effect/Diagnostics.hs:151]: I believe we're using `base >= 4.12 && < 4.14`; `NE.singleton` was added in 4.15

File: src/App/Fossa/Main.hs [hunk 3]
```diff
@@ -1,4 +1,93 @@
...
+      [ decodeSubCommand Analyze.analyzeSubCommand
+      , decodeSubCommand Test.testSubCommand
+      , decodeSubCommand Report.reportSubCommand
+      , decodeSubCommand Container.containerSubCommand
+      , decodeSubCommand VPS.vpsSubCommand
+      , decodeSubCommand ListTargets.listSubCommand
+      , decodeSubCommand Embed.dumpSubCommand
+      , decodeSubCommand LinkBins.linkBinsSubCommand
+      , decodeSubCommand Log4j.log4jSubCommand
+      , initCommand
```
[the remark this answers is not in the document]
Comment (scruffystuffs) [reply src/App/Fossa/Main.hs]: Yep, that was the problem.  Using separate subparsers and combining with `<|>` solved it.

File: src/App/Fossa/Main.hs [hunk 4]
```diff
@@ -1,4 +1,93 @@
...
+progData = fullDesc <> header "fossa-cli - Flexible, performant dependency analysis"
+
+subcommands :: Parser (IO ())
+subcommands =
+  hsubparser $
+    mconcat
+      [ decodeSubCommand Analyze.analyzeSubCommand
+      , decodeSubCommand Test.testSubCommand
+      , decodeSubCommand Report.reportSubCommand
+      , decodeSubCommand Container.containerSubCommand
```
[the remark this answers is not in the document]
Comment (scruffystuffs) [reply src/App/Fossa/Main.hs]: Lol, HLS threw that in there, even thought that would literally almost never be right.

**Pull request merged**